pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -108,7 +108,11 @@ def _process_stream(arg):
         if self._keepfiles:
             print("Solver log file: "+self._log_file)
 
-        if len(self._solver_model.objective.get_quadratic()) != 0:
+        obj_repn = generate_canonical_repn(self._objective.expr)
+        obj_degree = canonical_degree(obj_repn)
```

Comment:
This seems like an awful lot of work.  Why not just:
```
obj_degree = self._objective.expr.polynomial_degree()
if obj_degree is None or obj_degree > 2:
    raise #...
```
(Note the check for None - in Python 3, comparing None with an `int` raises an exception)

Reply:
Good suggestion! Thanks John!